src/OrchardCore/OrchardCore.DisplayManagement/Shapes/CoreShapes.cs

```diff
@@ -49,22 +43,29 @@ public void PlaceChildContent(dynamic Source, TextWriter Output)
                 return HtmlString.Empty;
             }
 
-            string listTagName = null;
+            var listTagName = "ul";
 
-            if (Tag != "-")
+            if (Shape.Properties.TryGetValue("TagName", out var value))
```

Comment:
As in other places, better to use `(...&& value is string valueString)` in place of using `ToString()` below.